### PR TITLE
[SALTO-1456] Delete DB on clean

### PR DESCRIPTION
**Pull request by yanaise**
---

---
When salto clean is called (unless cache is not deleted), database will be deleted from file system

### CODE REVIEW

File: packages/core/src/local-workspace/remote_map.ts [hunk 1]
```diff
@@ -22,6 +22,8 @@ import { remoteMap } from '@salto-io/workspace'
 import { collections, promises, values } from '@salto-io/lowerdash'
 import type rocksdb from '@salto-io/rocksdb'
 import path from 'path'
+import fs from 'fs'
```
Comment (ori-moisis) [inline packages/core/src/local-workspace/remote_map.ts:25]: is there a reason not to use `fileUtils` like everything else in our codebase?

File: packages/core/src/local-workspace/remote_map.ts [hunk 2]
```diff
@@ -51,6 +52,8 @@ const getRemoteDbImpl = (): any => {
   return rocksdbImpl
 }
 
+const getTmpLocationForLoc = (location: string): string => path.join(location, TMP_DB_DIR)
```
Comment (ori-moisis) [inline packages/core/src/local-workspace/remote_map.ts:55]: nit
```suggestion
const getDBTmpDir = (location: string): string => path.join(location, TMP_DB_DIR)
```

File: packages/core/src/local-workspace/remote_map.ts [hunk 3]
```diff
@@ -176,9 +179,13 @@ const persistentDBConnections: Record<string, Promise<rocksdb>> = {}
 const tmpDBConnections: Record<string, Record<string, Promise<rocksdb>>> = {}
 let currentConnectionsCount = 0
 
-const closeConnection = async (location: string, connection: Promise<rocksdb>): Promise<void> => {
+const closeConnection = async (location: string, connection: Promise<rocksdb>,
+  deleteDB = false): Promise<void> => {
```
Comment (ori-moisis) [inline packages/core/src/local-workspace/remote_map.ts:183]: nit
why is this coupled with closing the connection?
seems like a strange api for deleting, don't you think?

File: packages/core/src/local-workspace/remote_map.ts [hunk 4]
```diff
@@ -176,9 +179,13 @@ const persistentDBConnections: Record<string, Promise<rocksdb>> = {}
...
   const dbConnection = await connection
   await promisify(dbConnection.close.bind(dbConnection))()
+  if (deleteDB === true) {
+    await promisify(getRemoteDbImpl().destroy.bind(getRemoteDbImpl(), location))()
```
Comment (ori-moisis) [inline packages/core/src/local-workspace/remote_map.ts:187]: not sure what `getRemoteDbImpl` is, but should we be calling it twice here?

File: packages/core/src/local-workspace/remote_map.ts [hunk 5]
```diff
@@ -204,6 +211,18 @@ export const closeAllRemoteMaps = async (): Promise<void> => {
   })
 }
 
+export const cleanDatabases = async (): Promise<void> => {
+  const persistentDBs = Object.entries(persistentDBConnections)
```
Comment (ori-moisis) [inline packages/core/src/local-workspace/remote_map.ts:215]: I have no idea what `persistentDBConnections` is and where it comes from
@roironn - please review this part

**Review by roironn — APPROVED** [review]
Approved pendding Ori's comments :)

File: packages/core/src/local-workspace/remote_map.ts (hunk 4 above)
Comment (yanaise) [reply packages/core/src/local-workspace/remote_map.ts:187]: it's a singleton method that gives you the rocksdb implementation. It doesn't recreate it if you call it twice, and I think putting it in a variable would be less readable

File: packages/core/src/local-workspace/remote_map.ts (hunk 5 above)
Comment (yanaise) [reply packages/core/src/local-workspace/remote_map.ts:215]: he reviewed this

**Pull request merged**